sktime/forecasting/base/adapters/_pytorchforecasting.py

```diff
@@ -0,0 +1,524 @@
+            deepcopy(validation_to_dataloader_params)
+            if validation_to_dataloader_params is not None
+            else {}
+        )
+        self.random_log_path = random_log_path
+        super().__init__()
+
+    @functools.cached_property
+    @abc.abstractmethod
+    def algorithm_class(self: "_PytorchForecastingAdapter"):
+        """Import underlying pytorch-forecasting algorithm class."""
+
+    @functools.cached_property
+    @abc.abstractmethod
+    def algorithm_parameters(self: "_PytorchForecastingAdapter") -> dict:
+        """Keyword parameters for the underlying pytorch-forecasting algorithm class.
+
+        Returns
+        -------
+        dict
+            keyword arguments for the underlying algorithm class
+
+        """
+
+    def _instantiate_model(self: "_PytorchForecastingAdapter", data):
+        """Instantiate the model."""
+        algorithm_instance = self.algorithm_class.from_dataset(
+            data,
+            **self.algorithm_parameters,
+            **self._model_params,
+        )
+        import lightning.pytorch as pl
+
+        if self.random_log_path:
+            if "logger" not in self._trainer_params.keys():
+                if "default_root_dir" not in self._trainer_params.keys():
+                    random_num = (
+                        hash(time.time_ns())
+                        + hash(self.algorithm_class)
+                        + hash(str(data.get_parameters()))
+                        + hash(randint(0, int(time.time())))
+                    )
+                    self._random_log_dir = (
+                        os.getcwd() + "/lightning_logs/" + str(abs(random_num))
+                    )
+                    self._trainer_params["default_root_dir"] = self._random_log_dir
+
+        trainer_instance = pl.Trainer(**self._trainer_params)
+        return algorithm_instance, trainer_instance
+
+    def _fit(
+        self: "_PytorchForecastingAdapter",
+        y: pd.DataFrame,
+        X: typing.Optional[pd.DataFrame],
+        fh: ForecastingHorizon,
+    ) -> "_PytorchForecastingAdapter":
+        """Fit forecaster to training data.
+
+        private _fit containing the core logic, called from fit
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        y : sktime time series object
+            guaranteed to have a single column/variable
+        fh : guaranteed to be ForecastingHorizon
+            The forecasting horizon with the steps ahead to to predict.
+        X : sktime time series object, optional (default=None)
+            guaranteed to have at least one column/variable
+            Exogeneous time series to fit to.
+
+        Returns
+        -------
+        self : _PytorchForecastingAdapter
+            reference to self
+        """
+        self._max_prediction_length = np.max(fh.to_relative(self.cutoff))
+        if not np.min(fh.to_relative(self.cutoff)) > 0:
```

Comment:
There is a method to check if the `ForecastingHorizon` is `out-of-sample` 
`fh.is_all_out_of_sample`

Reply:
Thanks, I will change it in [[ENH] DeepAR and NHiTS and refinements for pytorch-forecasting interface](https://github.com/sktime/sktime/pull/6551#top)
#6551.

Reply:
I would say, we should change it in this PR, because here is where the code is being added - we should use the idiomatic method from the get-go. So I would, more specifically, request to make the change in this PR, so it ends up not having to be changed later but being the right method from the start.

Reply:
Yeah, changes are added to this PR now.